### ISSUE TITLE
Port as Chrome/ium extension

### ISSUE THREAD

**Issue by Jak2k** [rugk/googerteller#2]
## Background

I can't use it in vivaldi

## Proposed solution

An extension for Chromium

## Alternatives

Removing the star 🤣

**Comment by rugk** [comment]
[As elaborated on Twitter](https://mobile.twitter.com/rugkme/status/1561901665468432384) I doubt this is easily possible.

I have got some tips to solve the issue with missing cors header that could work in Chrome/ium too, but the DNS api is just not supported in Chromium, so this is unlikely to happen soon.
See https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/dns/resolve#browser_compatibility

And that is essentially required to resolve Google IPs.

---

Even that were solved, there would be the usual remaks about how I would accept such a port:

As always, I refer to https://github.com/rugk/offline-qr-code/issues/59 for details on how to make it.

The TL;DR is: The code base must keep supporting the new WebExtension (promise-based) API of Firefox, but we can include polyfills in the Chrome/ium version. Whether all features really work is not clear, though.

**Comment by Jak2k** [comment]
I readed it already on mastodon.

What about resolving the dns with a request to a dns server (dns over https)?
Or maybee also recognize the google domains?

**Comment by Jak2k** [comment]
Firefox is superior.

**Comment by rugk** [comment]
Hehe, I agree (and your apparent change in opinion over the years is interesting), but as for such an extension, I think it would still not hurt, to have cross-browser-compatibility. I personally also would not put much work into it, especially as this extension is mostly intended as "fun experiment" and not something serious. So, IMHO, we can leave this issue open and tag it with "help wanted", in case someone cares to port this extension.

Technically, BTW, a "polyfill" using DNS over HTTPS for the DNS resolve API would very likely be possible. Though it would literally resolve _all_ hostnames again a user visits, which is a waste of resources and a potential privacy problem. In case of Firefox API, I guess, it's just returned from Firefox DNS cache and thus not a new request happens.
